ProjectBank/Server.Integration.Tests/ProjectTests.cs

```diff
@@ -0,0 +1,81 @@
+namespace Server.Integration.Tests;
+
+public class ProjectTests : IClassFixture<CustomWebApplicationFactory>
+{
+    private readonly HttpClient _client;
+    private readonly CustomWebApplicationFactory _factory;
+    public ProjectTests(CustomWebApplicationFactory factory)
+    {
+        _factory = factory;
+        _client = factory.CreateClient(new WebApplicationFactoryClientOptions
+        {
+            AllowAutoRedirect = false
+        });
+    }
+
+    [Fact]
+    public async Task Get_returns_projects()
+    {
+        var projects = await _client.GetFromJsonAsync<ProjectDTO[]>("/api/projects");
+
+        Assert.NotNull(projects);
+        Assert.True(projects.Length >= 3);
+        Assert.Contains(projects, p => p.Title == "Super Fun Project");
+    }
+
+    [Fact]
+    public async Task Get_returns_specific_project()
+    {
+        var project = await _client.GetFromJsonAsync<ProjectDTO>("/api/projects/1");
+
+        Assert.NotNull(project);
+        Assert.Equal(1, project.Id);
+        Assert.Equal("Super Fun Project", project.Title);
+    }
+
+    [Fact]
+    public async Task Post_returns_Created()
+    {
+        var project = new CreateProjectDTO
+        {
+            Title = "Nice project title",
+            Description = "A short project description",
+            SupervisorId = "1",
+            Tags = new List<string>() { "DMAT", "Algo" },
+        };
+
+
+        var response = await _client.PostAsJsonAsync("/api/projects", project);
+
+        Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+        Assert.Equal(new Uri("http://localhost/api/Projects/4"), response.Headers.Location);
+
+        var created = await response.Content.ReadFromJsonAsync<ProjectDTO>();
+
+        Assert.NotNull(created);
+        Assert.Equal(4, created.Id);
+        Assert.Equal("Nice project title", created.Title);

```

Comment:
Can we test the tags and supervisorid here?

Reply:
The test used to assert just as many things as the unit test, i argued that since the specific method is tested in unit test, and the purpose of integration testing is testing components together, that we needn't assert everything. I don't know if this is wrong to assume.